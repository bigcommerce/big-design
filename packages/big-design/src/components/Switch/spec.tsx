import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { Switch } from './index';

describe('render Switch', () => {
  test('checked', () => {
    const { container } = render(<Switch checked={true} onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('unchecked', () => {
    const { container } = render(<Switch checked={false} onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled checked', () => {
    const { container } = render(<Switch checked={true} disabled onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled unchecked', () => {
    const { container } = render(<Switch checked={false} disabled onChange={() => null} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('has correct value for checked', async () => {
    render(<Switch checked={true} data-testid="switch" onChange={() => null} />);

    const input = await screen.findByTestId<HTMLInputElement>('switch');

    expect(input).toBeChecked();
  });

  test('has correct value for unchecked', async () => {
    render(<Switch checked={false} data-testid="switch" onChange={() => null} />);

    const input = await screen.findByTestId<HTMLInputElement>('switch');

    expect(input).not.toBeChecked();
  });

  test('triggers onChange when clicking the checkbox', async () => {
    const onChange = jest.fn();

    render(<Switch checked={true} data-testid="switch" onChange={onChange} />);

    const checkbox = await screen.findByTestId<HTMLInputElement>('switch');

    await userEvent.click(checkbox);

    expect(onChange).toHaveBeenCalled();
  });

  test('forwards ref', () => {
    const ref = createRef<HTMLInputElement>();

    const { container } = render(<Switch data-testid="switch" ref={ref} />);
    const input = container.querySelector('input');

    expect(input).toBe(ref.current);
  });
});
