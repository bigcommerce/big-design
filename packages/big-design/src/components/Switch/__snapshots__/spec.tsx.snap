// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render Switch checked 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0:focus + label::before {
  box-shadow: 0px 0px 0px 4px rgba(60,100,244,0.2);
}

@media (min-width:0px) {

}

@media (min-width:720px) {

}

<input
  checked=""
  class="c0"
  data-testid="switch"
  id=":r0:"
  type="checkbox"
/>
`;

exports[`render Switch disabled checked 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0:focus + label::before {
  box-shadow: 0px 0px 0px 4px rgba(60,100,244,0.2);
}

@media (min-width:0px) {

}

@media (min-width:720px) {

}

<input
  checked=""
  class="c0"
  data-testid="switch"
  disabled=""
  id=":r2:"
  type="checkbox"
/>
`;

exports[`render Switch disabled unchecked 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0:focus + label::before {
  box-shadow: 0px 0px 0px 4px rgba(60,100,244,0.2);
}

@media (min-width:0px) {

}

@media (min-width:720px) {

}

<input
  class="c0"
  data-testid="switch"
  disabled=""
  id=":r3:"
  type="checkbox"
/>
`;

exports[`render Switch unchecked 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c0:focus + label::before {
  box-shadow: 0px 0px 0px 4px rgba(60,100,244,0.2);
}

@media (min-width:0px) {

}

@media (min-width:720px) {

}

<input
  class="c0"
  data-testid="switch"
  id=":r1:"
  type="checkbox"
/>
`;
